filebeat/input/filestream/internal/input-logfile/store.go

```diff
@@ -332,6 +332,7 @@ func (s *store) remove(key string) error {
 	if resource == nil {
 		return fmt.Errorf("resource '%s' not found", key)
 	}
+	resource.version++
```

Comment:
Please add a comment why we increase the version on `remove`.

Shall we also use a boolean flag to 'invalidate' the resource? Once invalidated, we still allow the 'owner' to read and write to the resource, but updates will not be written to the persistent store anymore, no matter what happens to it. The only operation still allowed to be send to the persistent store is `Delete`.